### ISSUE TITLE
Kodi clean install, sync with Trakt - only TV shows/eps synced.  NO movies sync.

### ISSUE THREAD

**Issue by JoelAzaria** [razzeee/script.trakt#398]
Hey now.
I recently wiped kodi (for unrelated reasons) and so to keep my collection signed up w/ Trakt and installed the script.  I verified that movie and TV info appears at Trakt so I wiped Kodi and started fresh.
After I setup my addons and skin, etc, I installed Trakt, auth'd it and updated library.  All of my shows and eps have synced down to Kodi but not a single movie.

I tried various settings as per Trakt support but no joy.  Trakt support has suggested I post my log here to see if the dev can see what's going wrong.

Log here:  updated log coming

Please advise.  Thank you.

**Comment by JoelAzaria** [comment]
Updated log (sorry, previous log did not have trakt.script debugging on)
New log here: https://paste.ubuntu.com/25473944

debugging enabled on trakt script and on kodi.  did a library update, then did a reboot and a library update before posting this log.

Thanks.

**Comment by razzeee** [comment]
Your log would suggest that you can't connect to trakt at all

**Comment by JoelAzaria** [comment]
That is strange as it *has* synced my shows/eps and if I turn on the notifications on scrobble and sync I see messages telling me when it scrobbles and I see it start syncing both movies and TV (except the movie sync takes about a second to complete whereas TV shows takes longer)

I can try to re-auth the script and see if that helps.  I know my credentials are right because I can log in to the site with them and I successfully did the PIN thing so not sure why it would appear not to be able to connect.

**Comment by JoelAzaria** [comment]
FYI - I re-auth'd the script with a new PIN from the site.  No change.

BTW, could you tell me what line(s) show the failed login in the log above?  I had a look and couldn't see.  Thanks.

**Comment by razzeee** [comment]
See line 1175 for failed connection, not failed login.

Your not really getting a PIN inside kodi? You should not enter anything into kodi when getting auth. You should enter it into trakt.tv in a browser.

**Comment by JoelAzaria** [comment]
Sorry Razzeee should've been more clear.  I got a code in Kodi and entered it in the website as you say.  This was done correctly prior to me filing the bug report and done again above where I said I re-auth'd.  Results are the same.  Will review the log now see if I can make some sense of why the connection should fail.  No firewall nor anything else should be affecting so it is strange.  Report back shortly.

**Comment by razzeee** [comment]
But when you auth and enter the code from kodi into trakt, does kodi close the dialog automatically?

**Comment by JoelAzaria** [comment]
So I reviewed the log - not very verbose around the cause of the failure so I'm afraid I'm useless to help.   All I can add is that this is osmc/kodi running on an RPi3 and hardwired to my network.    There should be no rules in the firewall or elsewhere governing Kodi's reach to the Internet so the failure is strange.

The only other thing I could add is that while I agree with you and see the failure at 1175, further down the log you do see signs of (what looks to me like) accessing the trakt service and manipulating TV shows (but no mention of movies) so it looks to me like eventually it did succeed to reach the service or else I'm not sure wth I'm looking at.

Do you have any suggestions for how I can narrow down exactly what's failing or otherwise figure this out?

**Comment by JoelAzaria** [comment]
Yes, when I enter code in Trakt website it closes dialog automatically in Kodi.

**Comment by razzeee** [comment]
Whatever you see below is NOT this trakt addon. You have to look for `[script.trakt]` that's this addon. Not sure what the other one mentioning trakt is, just that it's done with `xbmcswift` which is a framework.

**Comment by JoelAzaria** [comment]
That's weird.  This is the only Trakt script/add-on I have.

edit-------

Just reviewed my full list of add-ons.  Confirm yours is the only reference to Trakt.  Not sure why/how anything else would be accessing trakt - I've not credentialed or auth'd any other software/device/add-on other than yours and this is a pretty new install of Kodi  (wiped about a week ago).

And as I said, it did restore my tv shows and when I turn on the notifications in the Trakt add-on script I see notifies on scrobble and on sync so your add-on is definitely functioning/doing something.

**Comment by razzeee** [comment]
Then do another logfile please. Try to sync maybe?

**Comment by JoelAzaria** [comment]
Hey Razzeee.

appreciate your help on this.  I assure I've tried syncing, rebooting and anything else I could think of to force a sync which is why I'm kind of at a dead-end.  I really have no idea how to diag this any further or even get a clue where to look.

Gimme a short bit and I'll gen another log.
Is it sufficient to turn on logging and just reboot Kodi (and wait for the startup stuff to run it's course) or do you want me to force a sync (and if so please specify from where/how to force.)

Again, thanks.

**Comment by JoelAzaria** [comment]
New logs: 
https://paste.ubuntu.com/25480217
https://paste.ubuntu.com/25480232

All I did was enable logging in the Trakt script, enable logging in Kodi->System and rebooted.  After reboot I left Kodi idle for 10 or 12 mins.  I expect that's plenty of time for whatever startup routines, library updates, etc. to complete.

I also changed Kodi Logfile Uploader to also include the old log so that's why two links above.  Looks like the 0232 is the old log and 0217 new.

**Comment by JoelAzaria** [comment]
Ok so now I'm really confused.  I tried to add a show on Trakt to my collection and then forced trakt script ot sync.  Show did not show up on Kodi.  Even though i get sync and scrobble messages from the trakt script on Kodi (including the red trakt script logo) it doesn't work.
More confusing is, if this indeed doesn't work, then how did I get my episodes back after the wipe?
Did it work one time then fail or did it never work?

Not sure what to make of it but reporting my observations nonetheless.

**Comment by razzeee** [comment]
Why would a show you add to your collection show up in kodi? That's not how it works unless your using illegal streaming addons...

**Comment by JoelAzaria** [comment]
Oh ok.  Then scratch that last part.  I'm still very new with Trakt.

**Comment by JoelAzaria** [comment]
Ok, so I tried the reverse - adding a show to Kodi.  I saw Kodi scan it against tvdb, I saw the trakt script notification of a sync and I saw the show appear in my collection on Trakt.
Then I tried the same with a movie and it too appears in my collection on Trakt.
However my Trakt collection of movies still hasn't synced down to Kodi.

J.

**Comment by rectifyer** [comment]
I think there is some confusion on what the collection is. The collection is movies and episodes you own, or in this case have in your local Kodi library. It should only sync from Kodi to Trakt. There isn't a need to sync your collection from Trakt to Kodi since the items actually need to exist in your Kodi library anyway. Does that make sense?

Trakt will 2 way sync your watched status. So, if you have a movie locally in your Kodi library that is watched on Trakt, that should mark it watched in Kodi as well.

**Comment by JoelAzaria** [comment]
Ok I get what you're saying but what about streaming?  There are streaming add-ons that scan shows and/or movies into the library alongside local content.   I expected this content to sync back down after the wipe.  Fact is, it did for TV but not for movies.

Further, I don't think it should be always the case that content have to be available to be kept in Kodi's library and/or synced to/from Trakt.
Consider the case of network storage or USB thumb/drives that may or not be online at any given moment.  The content should be in the library even though it's not necessarily online at the exact moment in time.

And there's also the case of streaming (not all streaming == piracy!) where an addon simply puts marker files in the movie or tv directories (in Kodi) so it appears in the library even though the video content is missing.  Trakt synced down all my shows that were like this so not understanding why movies should be different.

**Comment by rectifyer** [comment]
Anything related to streaming add-ons would be handled by them directly, it isn't something we provide help support for since those are separate add-ons from this one.

I get what you're saying about the library placeholders, but I don't think that is how Kodi is intending the local library feature to work. Are you sure it was Trakt that synced those episode placeholders to Kodi? That seems weird and not something this add-on would do. It would be very confusing to add something to your Trakt collection, then have a stub entry show up in Kodi that doesn't actually do anything since it's not actual video content. I don't think most users of the add-on would understand if that were the case.

**Comment by JoelAzaria** [comment]
@rectifyer I understand the separation between what is provided by add-on X and this one and I think I'm barking up the right tree.  I'm not looking for functions of an add-on I'm just looking to have trakt restore my library contents to how they were before I wiped.  I was expecting that by syncing my library to Trakt that it would sync it back how it was.  And it *did* for my TV shows and eps, just not for the movies.

wrt whether or not Kodi intends it to function that way I can't speak for them or anyone, I'm not even a coder but I can say that it works this way and that the dev for the add-on that I use for this doesn't call out any sort of hack that should break.  It's worked since Kodi 16 and through upgrades to 17, 17.1, .2, .3, etc.  
I don't think it's some sort of strange hack.  For purposes of appearing in the library Kodi doesn't seem to care exactly what's in a movie/show folder as long as it has an ID of some sort (imdb/tvdb/et al) to scrape and display it the movie/show as.

wrt whether this would confuse anyone I don't see how.  If they added it TO their library then they should know if/where/how the content is available to them.  If I add something to my lib because I have it on BD/DVD/Laserdisc/VHS/Betamax or if I add smth to my library because I know I can find it on Amazon Prime or Netflix or Hulu but I want in my Kodi lib because I spend 90+% of my viewing time IN Kodi then Trakt should sync that entry.  Either down to a wiped or new device or to any secondary/tertiary devices.  I don't see the problem - user added it to library, user expects it to *be* in library.
And since Trakt is supposed to *sync* library (between devices, devices and site, etc) it should sync ALL my library.   I don't think Trakt should decide whether an entry is 'valid' or a 'stub'.   Kodi put it in the lib, keep it and sync it.  Period.

That's imho of course but this is the more more sensible way it seems to me.

As far as whether it's the Trakt script that synced back my shows, yes I'm pretty close to certain.  I don't know how to be 1000% about it since the dev mentioned some framework thing in the debug logs but I did not install anything else that talks to Trakt nor did I authorize or give my login creds to any other software besides this script so afaik, nothing else should be able to sync or access anything on Trakt.  Only this script.

J.

**Comment by JoelAzaria** [comment]
Oh and just to clarify further, the add-on I use simply searches for something in IMDB and makes an entry in Kodi's library.  
So really it's an issue of Trakt + Kodi Library and right now it's only the movie side that is having a problem.

J.

**Comment by razzeee** [comment]
The other trakt addon is probably plugin.video.trakt_list_manager? Might
that be the case? Still not explaining why your series synced. Can you add
one series more and see if that gets added fine?

**Comment by JoelAzaria** [comment]
Hey @Razzeee 
wrt to other trakt add-on, definitely not trakt list manager.   Not installed, never was.   My best guess is maybe my skin or smth like extended info script.  My skin has widgets that show trakt data.  Maybe it's the widgets trying to pre-cache or smth?
As far as sync I'm certain it's your script.  There's nothing else on this system that's even able to and I verified by revoking and re-authing the script that it's the only thing syncing with the site.


Added another series.  Adds fine.

~~New wrinkle - it appears that series' which were fully watched (every ep) did not sync back down to my Kodi.  However series' with pending or unwatched eps did sync down to my Kodi box.  That's verified.
Untrue - I mistakenly had filtering on in Kodi~~
